codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/DirectedRubyCodegen.java

```diff
@@ -148,65 +156,77 @@ public void generateService(GenerateServiceDirective<GenerationContext, RubySett
 
     @Override
     public void generateStructure(GenerateStructureDirective<GenerationContext, RubySettings> directive) {
-        TypesGenerator typesGenerator = new TypesGenerator(directive.context());
-        typesGenerator.render();
-        typesGenerator.renderRbs();
-
-        ParamsGenerator paramsGenerator = new ParamsGenerator(directive.context());
-        paramsGenerator.render();
-
-        ValidatorsGenerator validatorsGenerator = new ValidatorsGenerator(directive.context());
-        validatorsGenerator.render();
-
-        GenerationContext context = directive.context();
-
-        if (directive.context().protocolGenerator().isPresent()) {
-            ProtocolGenerator generator = directive.context().protocolGenerator().get();
-            generator.generateBuilders(directive.context());
-            generator.generateParsers(directive.context());
-            generator.generateStubs(directive.context());
-        }
-
-        WaitersGenerator waitersGenerator = new WaitersGenerator(context);
-        waitersGenerator.render();
-        waitersGenerator.renderRbs();
-
-        PaginatorsGenerator paginatorsGenerator = new PaginatorsGenerator(context);
-        paginatorsGenerator.render();
-        paginatorsGenerator.renderRbs();
+        typeShapes.add(directive.shape());
     }
 
     @Override
     public void generateError(GenerateErrorDirective<GenerationContext, RubySettings> directive) {
         if (directive.context().protocolGenerator().isPresent()) {
             directive.context().protocolGenerator().get().generateErrors(directive.context());
         }
+        typeShapes.add(directive.shape());
```

Comment:
Ditto - why do we care about typeShapes in "generateError" ?

Reply:
Because `generateError` returns the Structure shape of an error.